src/HDF5.jl

```diff
@@ -1087,8 +1087,22 @@ dataspace(sz1::Int, sz2::Int, sz3::Int...; max_dims::Union{Dims,Tuple{}}=()) = _
 
 
 function get_dims(dspace::Dataspace)
-    dims = h5s_get_simple_extent_dims(dspace)
-    return tuple(reverse!(dims[1])...), tuple(reverse!(dims[2])...)
+    h5_dims, h5_maxdims = h5s_get_simple_extent_dims(dspace)
+    # reverse dimensions since hdf5 uses C-style order
+    N = length(h5_dims)
+    dims = ntuple(i -> Int(h5_dims[N-i+1]), N)
+    maxdims = ntuple(i -> h5_maxdims[N-i+1] % Int, N) # allows max_dims to be specified as -1 without triggering an overflow
```

Comment:
```suggestion
    dims = ntuple(i -> @inbound Int(h5_dims[N-i+1]), N)
    maxdims = ntuple(i -> @inbounds h5_maxdims[N-i+1] % Int, N) # allows max_dims to be specified as -1 without triggering an overflow
```

I see this reducing the LLVM and native code line counts by about half.